### DISCUSSION TITLE
Azure Static Web - deploy problem.

### DISCUSSION THREAD

**Discussion by maciejglowackimg** [gethinode/hinode#742] [Help] [answered]
I'm having an issue with Azure Static Web, and I'm getting the message 'Module "github.com/gethinode/hinode" is not compatible with this Hugo version; run "hugo mod graph" for more information.' I'm struggling with the problem. My Hugo version is hugo v0.112.5-48e33e10af2feb2a0e1c32d04880ed016619f2d8+extended, linux/amd64, BuildDate=2023-05-29T07:27:36Z, VendorInfo=gohugoio.
<img width="992" alt="Zrzut ekranu 2024-02-2 o 14 12 38" src="https://github.com/gethinode/hinode/assets/131944141/23cbcf09-e0e6-49e7-9c8c-5f6ba08894ee">

**Comment by markdumay** [comment]
Hinode requires [Hugo v0.120.0](https://github.com/gohugoio/hugo/releases/tag/v0.120.0) or later. It references the function [hugo.IsServer](https://gohugo.io/methods/site/isserver/) among other things.

**Reply by maciejglowackimg** [reply]
@markdumay Thank you for your answer, see line 150 is Hugo 0.121.5.

**Reply by markdumay** [reply]
Your screenshot refers to `hugo v0.112.5-4833e10af2feb2a0e1c32d04880ed016619f2d8+extended`on line 150?

**Reply by maciejglowackimg** [reply]
@markdumay  you're right... my mistake , do you know where I can force the installation of a specific version of Hugo?

**Reply by markdumay** [reply]
It depends on your build environment. Do you use GitHub actions to build and deploy your site? Could you share your action / deployment script if possible?

**Reply by maciejglowackimg** [reply]
@markdumay 
`name: Azure Static Web Apps CI/CD

on:
  push:
    branches:
      - main
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - main

jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: true
          lfs: false
      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_BLACK_MUD_0E0891610 }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: "upload"
          ###### Repository/Build Configurations - These values can be configured to match your app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: "/" # App source code path
          api_location: "" # Api source code path - optional
          output_location: "public" # Built app content directory - optional
          ###### End of Repository/Build Configurations ######

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_BLACK_MUD_0E0891610 }}
          action: "close"`

**Reply by markdumay** [reply, marked as answer]
Since you are using `Azure/static-web-apps-deploy@v1`, you might want to check the Azure docs: https://learn.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables.

From the docs:

```
with:
  azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN }}
  repo_token: ${{ secrets.GITHUB_TOKEN }}
  action: 'upload'
  app_location: 'src'
  api_location: 'api'
  output_location: 'public'
env: # Add environment variables here
  HUGO_VERSION: 0.58.0
```

I haven't tested this myself, so I'm not sure if the latest Hugo version is available.

**Reply by maciejglowackimg** [reply]
Works with Hugo 0.121.0. Thank you for your help! Actually, it was my oversight.
<img width="1082" alt="Zrzut ekranu 2024-02-2 o 19 46 10" src="https://github.com/gethinode/hinode/assets/131944141/d4ded093-40c5-4183-869e-7bd2604b3c9b">
